packages/fxa-payments-server/src/components/fields/index.tsx

```diff
@@ -343,7 +343,10 @@ export const StripeElement = (props: WrappedStripeElementProps) => {
   );
 };
 
-type CheckboxProps = { onValidate?: OnValidateFunction } & FieldProps &
+type CheckboxProps = {
+  onClick?: (event: React.MouseEvent<HTMLInputElement>) => void;
```

Comment:
You don't need to add this explicitly to the `CheckboxProps`, as it is covered by:
```
React.DetailedHTMLProps<
  React.InputHTMLAttributes<HTMLInputElement>,
  HTMLInputElement
>
```
You shouldn't need to modify this file at all actually. The `onClick` handler gets passed down implicitly if it's there as an optional prop. This is demonstrated already in the `onClick` prop for the consent checkbox in `SubscriptionUpgrade`.